sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClient.scala

```diff
@@ -26,6 +26,11 @@ import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.types.StructType
 
 
+private[hive] trait RawHiveTable {
+  def rawTable: Object
```

Comment:
Why you choose to have this one be `Object` then does instance of later `rawHiveTable.rawTable.asInstanceOf[HiveTable]`? Instead, it can just be a `HiveTable`?